src/dll/Systems/ScriptSystem.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "Hook.hpp"
+#include "ISystem.hpp"
+#include "Paths.hpp"
+#include "PluginBase.hpp"
+
+class ScriptSystem : public ISystem
+{
+public:
+    ScriptSystem(const Paths& aPaths);
+
+    ESystemType GetType() final;
+
+    void Startup() final;
+    void Shutdown() final;
+
+    bool Add(std::shared_ptr<PluginBase> aPlugin, const char* path);
+    std::wstring GetRedModArgs();
+    void SetScriptsBlobPath(RED4ext::CString*);
+    RED4ext::CString* GetScriptsBlobPath();
+    void SetUsingRedmod(bool);
+    bool IsUsingRedmod();
+    std::wstring CreatePathsFile();
+
+private:
+    bool _Add(std::shared_ptr<PluginBase> aPlugin, std::filesystem::path* path);
```

Comment:
```suggestion
    bool Add(std::shared_ptr<PluginBase> aPlugin, std::filesystem::path* path);
```
Please move it after `using`.